Source/Scene/Model.js

```diff
@@ -1512,9 +1513,9 @@ define([
         }
 
         // Modify draw program to receive shadows
-        if (frameState.shadowsEnabled && model.receiveShadows) {
+        if (shadowsEnabled && model.receiveShadows) {
             drawVS = ShadowMapShader.createShadowReceiveVertexShader(drawVS);
-            drawFS = ShadowMapShader.createShadowReceiveFragmentShader(drawFS, context);
+            drawFS = ShadowMapShader.createShadowReceiveFragmentShader(drawFS, frameState, 'v_normal');
```

Comment:
Ah right.  Hmm, I guess we can only do this reliably when we generate the shader client-side.  Still put a TODO here so we can come up with the most robust approach.